File: py/distance_measurement.py
import RPi.GPIO as GPIO
import time
from setup import *


def distance_measurement():
    GPIO.setup(trig_right, GPIO.OUT)
    GPIO.setup(echo_right, GPIO.IN)
    GPIO.output(trig_right, False)
    time.sleep(0.00001)
    GPIO.output(trig_right, True)
    time.sleep(0.00001)
    GPIO.output(trig_right, False)

    pulse_start = GPIO.wait_for_edge(echo_right, GPIO.RISING, timeout=500)
    if pulse_start is None:
        return float('inf')

    pulse_end = GPIO.wait_for_edge(echo_right, GPIO.FALLING, timeout=500)
    if pulse_end is None:
        return float('inf')

    pulse_duration = pulse_end - pulse_start
    distance = pulse_duration * 17150
    distance = round(distance, 2)
    return distance


def distance_monitoring_thread():
    while True:
        distance = distance_measurement()
        print("Distance: {} cm".format(distance))
        time.sleep(0.1)

File: js/distance_measurement.js
const pigpio = require('pigpio-client')().then((pi) => {
  const trigRight = 5
  const echoRight = 6

  pi.gpio(trigRight).modeSet('output');
  pi.gpio(echoRight).modeSet('input');

  console.log('Distance measurement is in progress...');

  pi.gpio(trigRight).digitalWrite(0);
  console.log('Sensor settles');
  setTimeout(() => {
    pi.gpio(trigRight).digitalWrite(1);
    setTimeout(() => {
      pi.gpio(trigRight).digitalWrite(0);

      let pulseStart;
      while (pi.gpio(echoRight).digitalRead() === 0) {
        pulseStart = process.hrtime.bigint();
      }

      let pulseEnd;
      while (pi.gpio(echoRight).digitalRead() === 1) {
        pulseEnd = process.hrtime.bigint();
      }

      const pulseDuration = Number(pulseEnd - pulseStart) / 1e6;
      const distance = pulseDuration * 17150;
      console.log(`Distance: ${distance.toFixed(2)} cm`);

      pi.end();
    }, 10);
  }, 2000);
});
